app/src/main/java/de/tum/in/tumcampusapp/component/ui/openinghour/OpeningApi.kt

```diff
@@ -0,0 +1,23 @@
+package de.tum.`in`.tumcampusapp.component.ui.openinghour
+
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+
+interface OpeningApi {
+    companion object {
+        private const val BASE_URL = "https://app.tum.de/"
+
+        fun buildOpeningHours(): OpeningApi {
+            val retrofit = Retrofit.Builder()
+                    .addCallAdapterFactory(
+                            RxJava2CallAdapterFactory.create())
+                    .addConverterFactory(
+                            GsonConverterFactory.create())
+                    .baseUrl(BASE_URL)
+                    .build()
+
+            return retrofit.create(OpeningApi::class.java)
+        }
+    }
```

Comment:
You don't need to build your own API here - we already have all of this.

Reply:
Kannst du mir in der Zukunft ein Link Schicken, wenn du ueber eine File redest.

Reply:
[But this is not the Endpoint you created](https://github.com/i-walker/Campus-Android/blob/e65e242c65ef7d479910547694338c1e16ac9e7e/app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeClient.java#L74)

Reply:
Retrofit generates all paths for all interface defintions. You don't need to create one retrofit client per endpoint. You can simply use the already initialized instance variable (singleton pattern): https://github.com/TUM-Dev/Campus-Android/blob/master/app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeClient.java#L103